engine/client/ref_common.c

```diff
@@ -659,23 +659,27 @@ qboolean R_Init( void )
 
 	R_CollectRendererNames();
 
-	// command line have priority
-	if( !Sys_GetParmFromCmdLine( "-ref", refopt ) )
-	{
+	// Priority:
+	// 1. Command line `-ref` argument.
+	// 2. `ref_dll` cvar.
+	// 3. Detected renderers in `DEFAULT_RENDERERS` order.
+	requested[0] = '\0';
+	if( !Sys_GetParmFromCmdLine( "-ref", requested ) && COM_CheckString( r_refdll->string ) )
 		// r_refdll is set to empty by default, so we can change hardcoded defaults just in case
-		Q_strncpy( refopt, COM_CheckString( r_refdll->string ) ?
-			r_refdll->string : DEFAULT_ACCELERATED_RENDERER, sizeof( refopt ) );
-	}
+		Q_strncpy( requested, r_refdll->string, sizeof( requested ) );
 
-	if( !(success = R_LoadRenderer( refopt )))
+	if ( requested[0] )
+		success = R_LoadRenderer( requested );
+
+	if( !success )
 	{
 		int i;
 
 		// cycle through renderers that we collected in CollectRendererNames
 		for( i = 0; i < ref.numRenderers; i++ )
 		{
 			// skip renderer that was requested but failed to load
-			if( Q_strcmp( refopt, ref.shortNames[i] ))
+			if( !Q_strcmp( requested, ref.shortNames[i] ) )
```

Comment:
@a1batross This is also a bugfix: Q_strcmp returns 0 when strings *are* equal

Reply:
Damn, right.